### PR TITLE
nektarpp: update to 5.3.0

### PR DESCRIPTION
**Pull request by mdave**
#### Description

This PR updates `nektarpp` to 5.3.0. Additionally:
- it removes two patches that have been upstreamed
- reduces third-party requirements due to the addition of the `tetgen` and `triangle` packages
- bumps the version of `boost` used to 1.76
- adds new Python versions for 3.10 and 3.11

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 13.2.1 22D68 arm64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: science/nektarpp/Portfile
Comment (reneeotten) [inline science/nektarpp/Portfile:18]: this is the default and can thus be removed

File: science/nektarpp/Portfile
Comment (reneeotten) [inline science/nektarpp/Portfile:32]: it looks like there is some additional clean-up possible now that there is only one distfile left and likely the standard from the `gitlab` PG would suffice?

File: science/nektarpp/Portfile
Comment (reneeotten) [inline science/nektarpp/Portfile:34]: still needed?

File: science/nektarpp/Portfile
Comment (reneeotten) [inline science/nektarpp/Portfile:35]: still needed?

File: science/nektarpp/Portfile
Comment (reneeotten) [inline science/nektarpp/Portfile:37]: file name shouldn't be required here anymore

File: science/nektarpp/Portfile
Comment (reneeotten) [inline science/nektarpp/Portfile:44]: still needed?

**Review by reneeotten — CHANGES_REQUESTED** [review]

**Comment by reneeotten** [comment]
@mdave thanks for the update! I've left a few comments above; also please check/fix the CI failure(s)

**Comment by catap** [comment]
I've fixed that mess with OneTBB and TBB at https://github.com/macports/macports-ports/pull/16915 which is blocked by only one PR https://github.com/macports/macports-ports/pull/16893

As soon as it merged I'll push rebased version of https://github.com/macports/macports-ports/pull/16915 which clean it up and a lot of similar issues.

File: science/nektarpp/Portfile
Comment (mdave) [reply science/nektarpp/Portfile:17]: Have removed (and also removed comments around 1.71)

File: science/nektarpp/Portfile
Comment (mdave) [reply science/nektarpp/Portfile:31]: I've left this in place for now, since if one of the `mpi` variants is enabled then we do need a third-party source that is not currently packaged (gsmpi). I have started to look at this, but will be a bit of work to complete it.

File: science/nektarpp/Portfile
Comment (mdave) [reply science/nektarpp/Portfile:33]: See above

File: science/nektarpp/Portfile
Comment (mdave) [reply science/nektarpp/Portfile:34]: See above

File: science/nektarpp/Portfile
Comment (mdave) [reply science/nektarpp/Portfile:36]: See above (I think?)

File: science/nektarpp/Portfile
Comment (mdave) [reply science/nektarpp/Portfile:43]: See above

**Comment by mdave** [comment]
@reneeotten Thanks for the review, I've pushed a few changes to hopefully address these.

@ryandesign Thanks for highlighting, I assumed that with `onetbb` changes there was some issue. We don't directly use `onetbb`, just appears through `opencascade`.

@catap Thanks for your efforts on the `onetbb` port!

File: science/nektarpp/Portfile
Comment (reneeotten) [reply science/nektarpp/Portfile:17]: I was talking about the `platforms` field, that's the default and should be removed. Please leave the `revision 0` line

**Comment by mdave** [comment]
Apologies I never made final change here for the `platforms` line that @reneeotten requested! Since 5.4.0 has now been released, I have applied the change above, opened a new PR for this and will now close this one.

**Pull request closed without merging**
